### PR TITLE
FEAT: History Functionality

### PR DESCRIPTION
**Pull request by fahd23**
Added History Functionality

### CODE REVIEW

File: src/context/Data-context.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { createContext, useReducer, useContext, useState } from "react";
 import { likeReducer } from "reducer/like-reducer";
 import { watchlaterReducer } from "reducer/watchLater-reducer";
+import { historyReducer } from "reducer/history-reducer";
```
Comment (t007rushi) [inline src/context/Data-context.js:4]: use index file for barrel export

**Review by t007rushi — APPROVED** [review]
Looks good to merge

**Review by JantuDeb — APPROVED** [review]
This commit is looking fine, code is structured and clean

**Review by mariyasada — APPROVED** [review]
the code structure is good.good to merge

**Pull request merged**